build.nix

```diff
@@ -149,6 +155,17 @@ with rec
             runHook postCheck
           '';
 
+
+        docPhase = lib.optionalString doDoc ''
+          runHook preDoc
+
+          echo "Running doc command:"
+          echo "  ${cargoDoc}"
+          CARGO_TARGET_DIR=target_doc ${cargoDoc}
```

Comment:
Do you know what the doc layout looks like in the `target` dir? Maybe we should just share the same `CARGO_TARGET_DIR` as the rest of the build (to avoid weird rebuilds) and simply copy out the doc-related files